wagtail/admin/views/pages/history.py

```diff
@@ -49,13 +48,21 @@ class WorkflowHistoryDetailView(
     workflow_history_url_name = "wagtailadmin_pages:workflow_history"
 
 
-@method_decorator(user_passes_test(user_has_any_page_permission), name="dispatch")
 class PageHistoryView(history.HistoryView):
     template_name = "wagtailadmin/pages/history.html"
     page_title = gettext_lazy("Page history")
     filterset_class = PageHistoryFilterSet
     model = Page
     pk_url_kwarg = "page_id"
+    permission_policy = page_permission_policy
+    any_permission_required = {
+        "add",
+        "change",
+        "publish",
+        "bulk_delete",
+        "lock",
+        "unlock",
+    }
```

Comment:
We should probably check for `PagePermissionTester.can_view_revisions()` instead, but for now I opted to just have our `PermissionCheckedMixin`-equivalent of the original decorator, just like what we did for the page listing views.

When wagtail/rfcs#92 is implemented, we could revisit this.